stat/combin/combin.go

```diff
@@ -368,3 +367,259 @@ func SubFor(sub []int, idx int, dims []int) []int {
 	sub[len(sub)-1] = idx
 	return sub
 }
+
+// NumPermutations returns the number of permutations when selecting k
+// objects from a set of n objects when the selection order matters.
+// No check is made for overflow.
+//
+// NumPermutations panics if either n or k is negative, or if k is
+// greater than n.
+func NumPermutations(n, k int) int {
+	if n < 0 {
+		panic("combin: n is negative")
+	}
+	if k < 0 {
+		panic("combin: k is negative")
+	}
+	if k > n {
+		panic("combin: k is greater than n")
+	}
+	p := 1
+	for i := n - k + 1; i <= n; i++ {
+		p *= i
```

Comment:
Should return -1 if we overflow?

Reply:
The comment explicitly states that there is no check for overflow, and this agrees with `Binomial`. I think this is how this PR should be, but feel free to file an issue

Reply:
OK